### DISCUSSION TITLE
module 'tinytuya' has no attribute 'Cloud'

### DISCUSSION THREAD

**Discussion by dpoitou-tlse** [jasonacox/tinytuya#339] [Q&A]
Hello
I try to launch my python script from a webpage. I use to do it form any other automations using jeedom.

It seems I have a permission issue but I do not undernstand how to solve it...

My code is
```
#!/usr/bin/python3.7
import tinytuya
import sys
print (sys.version)

# Connect to Tuya Cloud
c = tinytuya.Cloud()  # uses tinytuya.json 
```

When I launch my script from a web interface as www-data user I got this error :
```
3.7.3 (default, Oct 31 2022, 14:04:00)
[GCC 8.3.0]
Traceback (most recent call last):
  File "./Flood.PY", line 10, in 
    c = tinytuya.Cloud()  # uses tinytuya.json
AttributeError: module 'tinytuya' has no attribute 'Cloud'
```

If I run the same script as root user in a terminal it works well.
I tried to reinstall tinytuya with pip and checking that I did not used 'sudo' for installation
` python -m pip install tinytuya`
My installation is ok

The module seems to be correctly imported by my script but not the Cloud function...

Any idea ?
Best regards

**Comment by jasonacox** [comment]
See if python 3.7 is using the latest tinytuya:
```
/usr/bin/python3.7 -m tinytuya version
```
Should be close to 1.12.5 and if not:

```
/usr/bin/python3.7 -m pip install --upgrade tinytuya
```

**Reply by dpoitou-tlse** [reply]
Ok it works!
The version was 1.12.5 as root user but not as www-data user.
I made the update from this user and now it is  working.

Thanks

**Reply by jasonacox** [reply]
Awesome! Good fix.
